lib/iris/tests/unit/fileformats/netcdf/load_cube/load_cube__activate/__init__.py

```diff
@@ -0,0 +1,244 @@
+    actual Test_XXX class which also inherits unittest.TestCase.
+
+    Testcases are manufactured by the '_make_testcase_cdl' method.
+    These are based on a 'standard simple latlon grid' example.
+    Various kwargs control variations on this.
+
+    The 'run_testcase' method takes the '_make_testcase_cdl' kwargs and makes
+    a result cube (by: producing cdl, converting to netcdf, and loading).
+
+    The 'check_result' method performs various checks on the result, with
+    kwargs controlling the expected properties to be tested against.
+    This usage is *also* based on the 'standard simple latlon grid' example,
+    the kwargs specify expected differences from that.
+
+    Can also test with either the Pyke(rules) or non-Pyke (actions)
+    implementations (for now).
+
+    """
+
+    #
+    # "global" test settings
+    #
+
+    # whether to test 'rules' or 'actions' implementations
+    # TODO: remove when Pyke is gone
+    use_pyke = True
+
+    # whether to output various debug info
+    # TODO: ?possibly? remove when development is complete
+    debug = False
+
+    # whether to perform action in both ways and compare results.
+    compare_pyke_nonpyke = True
+
+    @classmethod
+    def setUpClass(cls):
+        # # Control which testing method we are applying.
+        # Create a temp directory for temp files.
+        cls.temp_dirpath = Path(tempfile.mkdtemp())
+
+    @classmethod
+    def tearDownClass(cls):
+        # Destroy a temp directory for temp files.
+        shutil.rmtree(cls.temp_dirpath)
+
+    def load_cube_from_cdl(self, cdl_string, cdl_path, nc_path):
+        """
+        Load the 'phenom' data variable in a CDL testcase, as a cube.
+
+        Using ncgen, CFReader and the _load_cube call.
+        Can use a genuine Pyke engine, or the actions mimic engine,
+        selected by `self.use_pyke`.
+
+        """
+        # Write the CDL to a file.
+        with open(cdl_path, "w") as f_out:
+            f_out.write(cdl_string)
+
+        # Create a netCDF file from the CDL file.
+        command = "ncgen -o {} {}".format(nc_path, cdl_path)
+        subprocess.check_call(command, shell=True)
+
+        # Simulate the inner part of the file reading process.
+        cf = CFReader(nc_path)
+        # Grab a data variable : FOR NOW, should be only 1
+        cf_var = list(cf.cf_group.data_variables.values())[0]
+        cf_var = cf.cf_group.data_variables["phenom"]
+
+        do_pyke = self.use_pyke or self.compare_pyke_nonpyke
+        do_nonpyke = not self.use_pyke or self.compare_pyke_nonpyke
+        if do_pyke:
+            pyke_engine = iris.fileformats.netcdf._pyke_kb_engine_real()
+        if do_nonpyke:
+            nonpyke_engine = iris.fileformats._nc_load_rules.engine.Engine()
+
+        iris.fileformats.netcdf.DEBUG = self.debug
+
+        # Call the main translation function to load a single cube.
+        def load_single_cube(engine):
```

Comment:
Would this make more sense to be named `load_specific_cube`? It seems like `_load_cube` is already loading a 'single' cube (i.e. the result is just one cube),  but this function if different in that it has already specified which cube it will load.

Reply:
I thin I may just call this 'translate_cube'.  
It is only needed to encapsulate the "call _load_cube and record formula info" operation, as we need to do that twice.